### DISCUSSION TITLE
Proper way to split a large directory into partitions respecting dirname and filename, and then create burnable iso for Blu-ray preserving the directory tree.

### DISCUSSION THREAD

**Discussion by amalgame21** [martymac/fpart#55] [Q&A]
I want to backup `/mnt/drive/photos` which have a very large directory tree like this:
```
......
/mnt/drive/photos/2015-02-15-Meeting-With-Friends/
/mnt/drive/photos/2015-02-15-Meeting-With-Friends/videos/
/mnt/drive/photos/2015-02-18-Family-Trip/
.......
/mnt/drive/photos/2020-04-26-Tim-Birthday/
/mnt/drive/photos/2020-06-10-Anthony-Birthday/
/mnt/drive/photos/2020-06-10-Anthony-Birthday/videos/
/mnt/drive/photos/2020-06-10-Anthony-Birthday/memo/
......
```

I would like to do 2 things:
1) Split the directory with `fpart` into multiple partitions, each of which contains less than 25GB of data (for 25GB Blu-ray disc), and the splitting should respect the dirname and filename in alphebatical order.
2) Create burnable Blu-ray ISOs using the partitions generated from the `fpart`.

After some trial and error, for 1),  I think it can be achieved by `find /mnt/drive/photos -type f | sort | fpart -L -s 23500m -i - -o photos`.
It generated multiple files with filename `photos.1` `photos.2` ... `photos.n`, I believed they are called partitions.
The partitions content seems to be sorted with alphabetical order.

Because some of the partitions exceed 25GB with the `-L -s 25g`, so I use `-L -s 23500m` instead.

After that, it is also challenging to make a burnable ISO. 
I guess the proper tool for this should be `mkisofs` or `xorrisofs`, because the `-path-list` flag of both tools seems to be designed to read the output generated by `fpart`.

What I want in the ISO is like this:
```
......
/2015-02-15-Meeting-With-Friends/
/2015-02-15-Meeting-With-Friends/videos/
/2015-02-18-Family-Trip/
.......
/2020-04-26-Tim-Birthday/
/2020-06-10-Anthony-Birthday/
/2020-06-10-Anthony-Birthday/videos/
/2020-06-10-Anthony-Birthday/memo/
......
```

I tested it with the command `mkisofs -o image1.iso -J -r -graft-points -path-list photos.1`
However, it seems that the `-graft-points` is not respected by `-path-list`, all the files in the partition `photos.1` are put to the root directory of ISO. It generated an error because some of them having the same filename.

`mkisofs -o image1.iso -J -r -graft-points /=/mnt/drive/photos -path-list photos.1`, also did not work. It did 2 things: 1) put ALL the directories inside `/mnt/drive/photos` into the root of ISO. 2) put all the files in the partition `photos.1` into the root of ISO.

I am pretty much stuck here after testing it for a whole day.
Am i using `fpart` or `mkisofs` improperly so they are not working together?

Please help!

**Comment by amalgame21** [comment]
Problem solved by manually editing the partition content into `-graft-points` format.
e.g. Change the content of the partition `photos.1` generated by `fpart` from
```
/mnt/drive/photos/2015-02-15-Meeting-With-Friends/img1.jpg
/mnt/drive/photos/2015-02-15-Meeting-With-Friends/img2.jpg
/mnt/drive/photos/2015-02-15-Meeting-With-Friends/img3.jpg
```
to
```
/2015-02-15-Meeting-With-Friends/img1.jpg=/mnt/drive/photos/2015-02-15-Meeting-With-Friends/img1.jpg
/2015-02-15-Meeting-With-Friends/img2.jpg=/mnt/drive/photos/2015-02-15-Meeting-With-Friends/img2.jpg
/2015-02-15-Meeting-With-Friends/img3.jpg=/mnt/drive/photos/2015-02-15-Meeting-With-Friends/img3.jpg

```
Then run `mkisofs -o image1.iso -J -r -graft-points -path-list photos.1`


Thank you very much for creating this tool!

**Comment by martymac** [comment]
Perfect, thanks for your feedback :)
